spec/rex/text/table_spec.rb

```diff
@@ -0,0 +1,158 @@
+require 'spec_helper'
+
+
+describe Rex::Text::Table do
+  it 'should space columns correctly' do
+    col_1_field = "A" * 5 
+    col_2_field = "B" * 50  
+    col_3_field = "C" * 15  
+
+    options = {
+      'Header'     => 'Header', 
+      'SearchTerm' => ['jim','bob'],
+      'Columns' => [
+        'Column 1',
+        'Column 2',
+        'Column 3'
+      ]
+    }
+    
+    tbl = Rex::Text::Table.new(options)
+
+    tbl << [
+      col_1_field,
+      col_2_field,
+      col_3_field
+    ] 
+   
+    expect(tbl.to_s).to eql <<~TABLE
+      Header
+      ======
+
+      Column 1  Column 2                                            Column 3
+      AAAAA     BBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBB  CCCCCCCCCCCCCCC
+    TABLE
+  end
+
+  it 'should apply field formatters correctly and increase column length' do
+    col_1_field = "A" * 5 
+    col_2_field = "B" * 50  
+    col_3_field = "C" * 15  
+
+    options = {
+      'Header'     => 'Header', 
+      'SearchTerm' => ['jim','bob'],
+      'Columns' => [
+        'Column 1',
+        {
+          column: 'Column 2',
+          formatters: [Formatter.new]
+        },
+        'Column 3'
+      ]
+    }
+    
+    tbl = Rex::Text::Table.new(options)
+
+    tbl << [
+      col_1_field,
+      col_2_field,
+      col_3_field
+    ] 
+   
+    expect(tbl.to_s).to eql <<~TABLE
+      Header
+      ======
+
+      Column 1  Column 2                                                              Column 3
+      AAAAA     IHAVEBEENFORMATTEDBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBB  CCCCCCCCCCCCCCC
+    TABLE
+  end
+
+
+
+  it 'should apply field stylers correctly and NOT increase column length' do
+    col_1_field = "A" * 5 
+    col_2_field = "B" * 50  
+    col_3_field = "C" * 15  
+
+    options = {
+      'Header'     => 'Header', 
+      'SearchTerm' => ['jim','bob'],
+      'Columns' => [
+        'Column 1',
+        {
+          column: 'Column 2',
+          stylers: [Styler.new]
+        },
```

Comment:
I prefer the symbol syntax here, but on reflection, what are your thoughts of being consistent with the existing string API?

```suggestion
        {
          'Column' => 'Column 2',
          'Stylers' => [Styler.new]
        },
```

Reply:
I like my efficiencies and I'd definitely prefer to use symbols for the performance improvement, but I doubt it's worth changing everything in the API and using a half & half approach is even worse. So I'd say keep everything consistent with the existing api and use hash rockets